impl/maven-impl/src/main/java/org/apache/maven/impl/AbstractSession.java

```diff
@@ -134,7 +134,7 @@ public AbstractSession(
             List<RemoteRepository> repositories,
             List<org.eclipse.aether.repository.RemoteRepository> resolverRepositories,
             Lookup lookup) {
-        this.session = nonNull(session, "session");
+        this.session = requireNonNull(session, "session");
         this.repositorySystem = repositorySystem;
```

Comment:
Kind of inconsequent. Why not this? Nullable too. Thus `session` not marked `@NonNull` so its kind of magic number?

We produce boilerplate for the only difference to have a little better error message. 

This is high cost for little, to almost no, return.